volumes/repository.go

```diff
@@ -80,38 +81,36 @@ func (r *Repository) newVolume(path string, writable bool) (*Volume, error) {
 }
 
 func (r *Repository) restore() error {
+	log.Debugf("Restoring volumes")
```

Comment:
This file looks to be full of refactoring not related to the fix.  Please don't refactor unrelated blocks in bug fixes.

Unless this is related to the fix for the bugs please remove this refactoring because there are a few bugs and areas where this code panics.